var/spack/repos/builtin/packages/virtuoso/package.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+from spack import *
+
+
+class Virtuoso(AutotoolsPackage):
+    """Virtuoso is a high-performance and scalable Multi-Model RDBMS,
+    Data Integration Middleware, Linked Data Deployment, and HTTP
+    Application Server Platform"""
+
+    homepage = "https://github.com/openlink/virtuoso-opensource"
+    url      = "https://github.com/openlink/virtuoso-opensource/archive/v7.2.5.1.tar.gz"
+
+    version('7.2.5.1', sha256='3e4807e94098b8265f8cf00867d1215bb1e9d0d274878e59a420742d2de471c2')
+
+    variant('dbpedia-vad', default=False, description='DBpedia vad package')
+    variant('demo-vad', default=False, description='Demo vad package')
+    variant('fct-vad', default=True, description='Facet Browser vad package')
+    variant('ods-vad', default=True, description='ODS vad package')
+    variant('sparqldemo-vad', default=False, description='Sparql Demo vad package')
+    variant('tutorial-vad', default=False, description='Tutorial vad package')
+    variant('isparql-vad', default=True, description='iSPARQL vad package')
+    variant('rdfmappers-vad', default=True, description='RDF Mappers vad package')
+
+    depends_on('autoconf', type='build')
+    depends_on('automake', type='build')
+    depends_on('libtool',  type='build')
+    depends_on('m4',       type='build')
+
+    depends_on('bison', type='build')
+    depends_on('flex', type='build')
+    depends_on('gperf')
+    depends_on('readline')
+    depends_on('openssl@0.9.8:1.1.99')
+
+    # Fix fail to include <rpc/types.h> problem
+    # https://github.com/openlink/virtuoso-opensource/commit/52b6f8ebe108c1ed86fb840305c5f5a9677228f5
+    patch('virt_rpc.patch')
+
+    # support openssl@1.1.1
+    # https://github.com/openlink/virtuoso-opensource/commit/713fa25b14457aa5127fac071830a2d20f4f968c
+    # https://github.com/openlink/virtuoso-opensource/commit/713fa25b14457aa5127fac071830a2d20f4f968c
+    # https://github.com/openlink/virtuoso-opensource/commit/713fa25b14457aa5127fac071830a2d20f4f968c
```

Comment:
I don't think we should add a 2,5k lines patch to add a feature. Can we either:
1. Add a new version that brings these changes, maybe tied to a git branch
2. Use the patch from https instead of adding code to Spack

? I would prefer 1 over 2 but both are imho better than the current state.

Reply:
1. Only `develop/7` branch includes these changes. This seems inadequate as source. The development is active.
2. There is very large difference between `develop/7` and latest tag `v7.2.5.1`. So patch fails.

Reply:
> Only develop/7 branch includes these changes. This seems inadequate as source

Why is this inadequate? Note that in a version directive you can either pick a branch (moving target) or a commit sha (fixed). I would really prefer if we drop patches and add another version with the feature you need.

Reply:
Changed to commit sha style.